noether_tpp/src/tool_path_planners/origin_generators.cpp

```diff
@@ -43,4 +58,11 @@ Eigen::Vector3d OffsetOriginGenerator::generate(const pcl::PolygonMesh& mesh) co
   return origin + offset_;
 }
 
+std::unique_ptr<OriginGenerator> OffsetOriginGenerator::clone() const
+{
+  if (generator_ == nullptr)
+    throw std::runtime_error("OffsetOriginGenerator has nullptr sub-generator");
+  return std::make_unique<OffsetOriginGenerator>(generator_->clone(), offset_);
+}
```

Comment:
It's well-established I just don't like exceptions, but it seems like just making a new one that also has a nullptr would be the expected behavior. *shrug*

Reply:
This function is gone. We can have this debate again later if it comes up